### DISCUSSION TITLE
Day-of-week abbreviations ending in periods not recognized

### DISCUSSION THREAD

**Discussion by neurolit** [alphapapa/org-ql#429] [Q&A] [answered]
Hi!

I don't know what I am doing wrong with org-ql-search.

I'm using Emacs 29.3 on Linux, with org-ql 0.8.6 and org-mode 9.7.3 (and no other package).

I've got a really simple tasks file (`/tmp/task.org`):
``` org
* TODO Eat some chips
SCHEDULED: <2024-06-12 mer.>
  :LOGBOOK:
  - State "TODO"       from              [2024-06-12 mer. 22:51]
  :END:
```

When I try `(org-ql-search "/tmp/task.org" '(todo))`, it works perfectly and displays the task:

```
  TODO Eat some chips  today 
```

However, if I try `(org-ql-search "/tmp/task.org" '(scheduled))`, nothing happens, the query buffer is empty. I think it should display the same task (as it is scheduled).

Where am I wrong?

**Comment by alphapapa** [comment]
Hi,

I'm guessing it has something to do with using the French for the day of the week there.  I don't know how that works internally in Org, as I haven't used that functionality myself, and I haven't heard from anyone with this problem before.

Looking at our code, I guess it's because the regexp in `org-ql-regexp-part-ts-date` expects the optional day name abbreviation to be alphabetical, i.e. it does not match the period.  Not having used other languages for the day name abbreviations, I didn't know that they could include periods.

In the Org manual, the relevant section seems to be:

```
   The function understands English month and weekday abbreviations.  If
you want to use un-abbreviated names and/or other languages, configure
the variables ‘parse-time-months’ and ‘parse-time-weekdays’.
```
So a potential workaround would be to remove the periods from the abbreviations I guess you've set in that variable (and then change your Org files accordingly).

Otherwise, if this represents a small incompatibility with Org itself, then it would be a bug to be fixed in Org QL.

Let me know what you think.  Thanks.

**Comment by neurolit** [comment]
Thank you for your answer. I'm not sure I understand: `(scheduled)` doesn't work, but `(ts)` does:

``` emacs-lisp
(org-ql-search "/tmp/task.org" '(ts :on -1))
```

gives, as expected:
```
  TODO Eat some chips  1d ago 

```

Moreover, I didn't change anything in Emacs or Org configuration for french abbreviations. Here is my `init.el`:

``` emacs-lisp
(defvar elpaca-installer-version 0.7)
(defvar elpaca-directory (expand-file-name "elpaca/" user-emacs-directory))
(defvar elpaca-builds-directory (expand-file-name "builds/" elpaca-directory))
(defvar elpaca-repos-directory (expand-file-name "repos/" elpaca-directory))
(defvar elpaca-order '(elpaca :repo "https://github.com/progfolio/elpaca.git"
                              :ref nil :depth 1
                              :files (:defaults "elpaca-test.el" (:exclude "extensions"))
                              :build (:not elpaca--activate-package)))
(let* ((repo  (expand-file-name "elpaca/" elpaca-repos-directory))
       (build (expand-file-name "elpaca/" elpaca-builds-directory))
       (order (cdr elpaca-order))
       (default-directory repo))
  (add-to-list 'load-path (if (file-exists-p build) build repo))
  (unless (file-exists-p repo)
    (make-directory repo t)
    (when (< emacs-major-version 28) (require 'subr-x))
    (condition-case-unless-debug err
        (if-let ((buffer (pop-to-buffer-same-window "*elpaca-bootstrap*"))
                 ((zerop (apply #'call-process `("git" nil ,buffer t "clone"
                                                 ,@(when-let ((depth (plist-get order :depth)))
                                                     (list (format "--depth=%d" depth) "--no-single-branch"))
                                                 ,(plist-get order :repo) ,repo))))
                 ((zerop (call-process "git" nil buffer t "checkout"
                                       (or (plist-get order :ref) "--"))))
                 (emacs (concat invocation-directory invocation-name))
                 ((zerop (call-process emacs nil buffer nil "-Q" "-L" "." "--batch"
                                       "--eval" "(byte-recompile-directory \".\" 0 'force)")))
                 ((require 'elpaca))
                 ((elpaca-generate-autoloads "elpaca" repo)))
            (progn (message "%s" (buffer-string)) (kill-buffer buffer))
          (error "%s" (with-current-buffer buffer (buffer-string))))
      ((error) (warn "%s" err) (delete-directory repo 'recursive))))
  (unless (require 'elpaca-autoloads nil t)
    (require 'elpaca)
    (elpaca-generate-autoloads "elpaca" repo)
    (load "./elpaca-autoloads")))
(add-hook 'after-init-hook #'elpaca-process-queues)
(elpaca `(,@elpaca-order))

;; Install use-package support
(elpaca elpaca-use-package
  ;; Enable :elpaca use-package keyword.
  (elpaca-use-package-mode)
  (setq use-package-verbose nil))
(elpaca-wait)

(use-package org
  :ensure ( org
            :branch "bugfix" ))

(use-package org-ql
  :ensure t
  :defer t)
```

Here is the value of `parse-time-weekdays`:
``` emacs-lisp
(("sun" . 0)
 ("mon" . 1)
 ("tue" . 2)
 ("wed" . 3)
 ("thu" . 4)
 ("fri" . 5)
 ("sat" . 6)
 ("sunday" . 0)
 ("monday" . 1)
 ("tuesday" . 2)
 ("wednesday" . 3)
 ("thursday" . 4)
 ("friday" . 5)
 ("saturday" . 6))
```

**Reply by alphapapa** [reply]
So where is the `mer.` in `<2024-06-12 mer.>` coming from?

**Reply by neurolit** [reply]
Seems to be by default (since my locale is LANG=fr_FR.utf8): when I use `C-c C-s` (bound to `org-schedule`) and choose a date, the day is in french in the timestamp.

**Reply by neurolit** [reply]
I've got an unsatisfying workaround, in order to use, as you suggested, english days of the week. By default, `system-time-locale` is nil. When I set if to `"C"`, `org-schedule` writes `Thu` instead of `jeu.`.

However, it would be better if org-ql accepted french days of week for `scheduled` (as it does for `ts`).

**Reply by alphapapa** [reply]
This still seems strange, because according to the Org manual, IIUC, `parse-time-weekdays` should cause the English abbreviations to be used.  But those aren't capitalized, either, so that doesn't appear to be quite accurate.

If you could find a way to simply prevent the periods from being added at the end of the abbreviation, it would solve the problem, I think.

Anyway, see https://github.com/alphapapa/org-ql/issues/432

**Comment by alphapapa** [comment, marked as answer]
This should be fixed on v0.8.7 and master now. Thanks.
